### ISSUE TITLE
D102 - false positive on overload with comment

### ISSUE THREAD

**Issue by DetachHead** [PyCQA/pydocstyle#635]
```py
from typing import overload


class Foo:
    """asdf"""

    @overload  # this comment causes the error
    def foo(self):  # error: D102
        ...

    @overload
    def foo(self, value: int):  # no error because the docstring in on the implementation
        ...

    def foo(self, value: int | None = None):
        """asdf"""

```
